### PR TITLE
add config.h to cmake

### PR DESCRIPTION
**Pull request by angainor**
... and cleanup the defines: `GHEX_USE_PMIX` and `GHEX_USE_UCP` are added to `config.h` and can be used in the code after including it.

### CODE REVIEW

File: benchmarks/transport/CMakeLists.txt [hunk 1]
```diff
@@ -11,23 +11,11 @@ if (GHEX_USE_UCP)
         add_executable(${_t} ${_t}_mt.cpp )
         target_compile_definitions(${_t} PRIVATE USE_HEAVY_CALLBACKS USE_RAW_SHARED_MESSAGE USE_POOL_ALLOCATOR)
         target_link_libraries(${_t} ghexlib)
-        if (GHEX_USE_PMIX)
```
Comment (bettiolm) [inline benchmarks/transport/CMakeLists.txt:14]: I see, so the main benefit would be to avoid this repetition when setting up the targets in Cmake, right?

Comment (angainor) [reply benchmarks/transport/CMakeLists.txt:14]: @bettiolm precisely. all config is is one place, so once you add a cmake variable, you will not need to go to all subdirectories and update the targets.

File: benchmarks/transport/CMakeLists.txt [hunk 2]
```diff
@@ -11,23 +11,11 @@ if (GHEX_USE_UCP)
...
-            target_compile_definitions(${_t} PRIVATE USE_PMIX)
-        endif()
-        if (GHEX_USE_UCP)
-            target_compile_definitions(${_t} PRIVATE USE_UCX)
```
Comment (mbianco) [inline benchmarks/transport/CMakeLists.txt:18]: One CMake way of doing it is to set `target_compile_definitions(${_t} PUBLIC USE_UCX)`. In this way, when you link with GHEX cmake will add these options to the compilation line and your object files. It is true that if these options do not change once the library is installed, then we do not need this and we can use the `config.h`

Comment (angainor) [reply benchmarks/transport/CMakeLists.txt:18]: @mbianco `config.h` is regenerated every time you change the cmake input files and run generate, and it is also only created in the build directory (not installed). The point of it is that you don't have to add individual `target_compile_definitions(${_t} PUBLIC ...` for each new config variable you implement. It makes shorter and simpler to maintain `CMakeList.txt` files, but also more compact compile command lines without lots of `-D` statements.

**Pull request closed without merging**